tests/integration/ledger/evmosd_suite_test.go

```diff
@@ -0,0 +1,168 @@
+package ledger_test
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/simapp/params"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/ethermint/crypto/ethsecp256k1"
+	"github.com/evmos/ethermint/tests"
+	"github.com/evmos/evmos/v10/app"
+	"github.com/evmos/evmos/v10/tests/integration/ledger/mocks"
+	"github.com/stretchr/testify/suite"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	"github.com/tendermint/tendermint/version"
+
+	cosmosledger "github.com/cosmos/cosmos-sdk/crypto/ledger"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	bankTypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	evm "github.com/evmos/ethermint/x/evm/types"
+	feemarkettypes "github.com/evmos/ethermint/x/feemarket/types"
+	evmoskeyring "github.com/evmos/evmos/v10/crypto/keyring"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmversion "github.com/tendermint/tendermint/proto/tendermint/version"
+	rpcclientmock "github.com/tendermint/tendermint/rpc/client/mock"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var s *LedgerTestSuite
+
+type LedgerTestSuite struct {
+	suite.Suite
+
+	app *app.Evmos
+	ctx sdk.Context
+
+	queryClient    bankTypes.QueryClient
+	queryClientEvm evm.QueryClient
+
+	ledger       *mocks.SECP256K1
+	accRetriever *mocks.AccountRetriever
+
+	ethAddr     common.Address
+	accAddr     sdk.AccAddress
+	signer      keyring.Signer
+	privKey     *ethsecp256k1.PrivKey
+	pubKey      types.PubKey
+	consAddress sdk.ConsAddress
+}
+
+func TestLedger(t *testing.T) {
+	s = new(LedgerTestSuite)
+	suite.Run(t, s)
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Evmosd Suite")
+}
+
+func (suite *LedgerTestSuite) SetupTest() {
+	var err error
+	suite.ledger = mocks.NewSECP256K1(s.T())
+	suite.privKey, err = ethsecp256k1.GenerateKey()
+	s.Require().NoError(err)
+	suite.pubKey = suite.privKey.PubKey()
```

Comment:
why do you need the pubkey?

Reply:
I need the pubkey to return it from mocked `GetAddressPubKeySECP256K1`